### DISCUSSION TITLE
Feature Request: Replace Queue with song and following songs

### DISCUSSION THREAD

**Discussion by cjohnsto-nz** [jcorporation/myMPD#600] [Ideas]
This is a bit specific, so bear with me.

This is how I have my playback settings configured.

![image](https://user-images.githubusercontent.com/21297202/139775546-81b96656-48d9-4097-a2a0-6a9f293921cd.png)

The result is that when I click on a song, it plays immediately. I'm not big on the queue system being the primary interaction. If i want to add to the queue, I will always want to do it by right clicking.

If I go into an album and select a song it will replace the entire queue with that song. My preference would be to have an option that emulates the behaviour of Spotify and other common media apps, where it will replace the queue with the song you have selected, and any following songs in the album, playlist. Effectively, the album or playlist would continue playing from where you selected.

Would this be possible?

PS. You're doing amazing work. I almost can't believe how responsive you've been :)

**Comment by jcorporation** [comment]
MPD behaves a bit different than "more modern" players as spotify. I do not like the spotify or the sonos way.

This could be done. But it could be also useful for append to queue and insert into queue after playing song. Insert after playing song is implemented for the v8.2.0 release. With this and your feature request there are many eventually too many options in the menu.

- Append song to queue
- Insert song after current song
- Replace queue
- Append song and following songs to queue
- Insert song and following songs to queue
- Replace queue with song and following songs
- Each song has an album tab with append, insert and replace too

Have you an idea how this could be designed without too long contextmenus?

**Comment by cjohnsto-nz** [comment]
That's a good point. It does become quite cluttered.

For my use case, I don't really see it being useful from the context menu. The occasions where I would want to append or insert all songs from a following point in an album would be few and far between.

I guess I'm looking for an "Autoplay" option, which adds the songs following only when you click on an invidual song to replace the queue.

Further to that, if I right clicked on a song and selected replace queue, I would expect it to only play that song, so an Autoplay feature in my mind would be specific to click and replace.

So maybe an option could be added call Autoplay that functions like this. Only applies to songs, not albums (if that's possible), and doesn't have an option to insert (because it kind of defeats the purpose)

Then it could be selected as the playback action for song click if desired.
________________________________

**Comment by cjohnsto-nz** [comment]
Maybe a better name would be "Play from selected"
________________________________
